juju/origin.py

```diff
@@ -92,30 +94,39 @@ def parse(s: str):
         elif len(p) == 2:
             track = p[0]
             risk = p[1]
+        elif len(p) == 3:
+            track = p[0]
+            risk = p[1]
+            branch = p[2]
```

Comment:
Python built-in would be more elegant:

```suggestion
        elif len(p) == 3:
            track, risk, branch = p
```

Reply:
Or, alternatively something like:

```py
match parts:
    ...
    case [track, risk, branch]:
        pass
```

Ofc. refactoring is only possible if the function is covered by unit tests.